### PR TITLE
#102 Add alternative for php pathinfo function is discouraged

### PR DESCRIPTION
**Pull request by diazwatson**
#102

### CODE REVIEW

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 1]
```diff
@@ -194,24 +194,24 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
...
+        '^is_file$' => '\Magento\Framework\Filesystem\Driver\File->isFile()',
         '^is_link$' => null,
-        '^is_readable$' => null,
-        '^is_writable$' => null,
-        '^is_writeable$' => null,
+        '^is_readable$' => '\Magento\Framework\Filesystem\Driver\File->isReadable()',
+        '^is_writable$' => 'Magento\Framework\Filesystem\Io\File->isWriteable()',
+        '^is_writeable$' => 'Magento\Framework\Filesystem\Io\File->isWriteable()',
         '^is_uploaded_file$' => null,
         '^glob$' => null,
```
Comment (ihor-sviziev) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]: This probably could be replaced to `\Magento\Framework\Filesystem\Glob::glob` like there:
https://github.com/magento/magento2/blob/0b968b881498d650aed7b52ed5d78a5833c5da62/lib/internal/Magento/Framework/App/Utility/Files.php#L285

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 2]
```diff
@@ -194,24 +194,24 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
         '^gettype$' => null,
         '^var_dump$' => null,
         '^tempnam$' => null,
-        '^realpath$' => null,
+        '^realpath$' => '\Magento\Framework\Filesystem\Driver\File->getRealPath()',
         '^linkinfo$' => null,
         '^lstat$' => null,
         '^stat$' => null,
```
Comment (ihor-sviziev) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]: `\Magento\Framework\Filesystem\Driver\File->stat`

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 3]
```diff
@@ -53,14 +53,14 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
         '^dcngettext$' => null,
         '^dgettext$' => null,
         '^dio_.*$' => null,
-        '^dirname$' => null,
+        '^dirname$' => 'Magento\Framework\Filesystem\Io\File->dirname()',
```
Comment (ihor-sviziev) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]:
```suggestion
        '^dirname$' => '\Magento\Framework\Filesystem\Io\File->dirname',
```

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 4]
```diff
@@ -53,14 +53,14 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
...
         '^dngettext$' => null,
         '^domxml_.*$' => null,
         '^fbsql_.*$' => null,
         '^fdf_add_doc_javascript$' => null,
         '^fdf_open$' => null,
-        '^fopen$' => null,
-        '^fclose$' => null,
+        '^fopen$' => '\Magento\Framework\Filesystem\Driver\File->fileOpen()',
```
Comment (ihor-sviziev) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]:
```suggestion
        '^fopen$' => '\Magento\Framework\Filesystem\Driver\File->fileOpen',
```

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 5]
```diff
@@ -53,14 +53,14 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
...
+        '^fclose$' => '\Magento\Framework\Filesystem\Driver\File->fileClose()',
```
Comment (ihor-sviziev) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]:
```suggestion
        '^fclose$' => '\Magento\Framework\Filesystem\Driver\File->fileClose',
```

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 6]
```diff
@@ -96,7 +96,7 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
         '^parse_str$' => null,
         '^parse_url$' => null,
         '^parsekit_compile_string$' => null,
-        '^pathinfo$' => null,
+        '^pathinfo$' => 'Magento\Framework\Filesystem\Io\File->getPathInfo()',
```
Comment (ihor-sviziev) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]:
```suggestion
        '^pathinfo$' => 'Magento\Framework\Filesystem\Io\File->getPathInfo',
```

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 7]
```diff
@@ -194,24 +194,24 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
         '^gettype$' => null,
         '^var_dump$' => null,
         '^tempnam$' => null,
-        '^realpath$' => null,
+        '^realpath$' => '\Magento\Framework\Filesystem\Driver\File->getRealPath()',
```
Comment (ihor-sviziev) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]:
```suggestion
        '^realpath$' => '\Magento\Framework\Filesystem\Driver\File->getRealPath',
```

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 8]
```diff
@@ -194,24 +194,24 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
...
         '^linkinfo$' => null,
         '^lstat$' => null,
         '^stat$' => null,
         '^lchgrp$' => null,
         '^lchown$' => null,
         '^show_source$' => null,
-        '^is_dir$' => null,
+        '^is_dir$' => '\Magento\Framework\Filesystem\Driver\File->isDirectory()',
```
Comment (ihor-sviziev) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]:
```suggestion
        '^is_dir$' => '\Magento\Framework\Filesystem\Driver\File->isDirectory',
```

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 9]
```diff
@@ -194,24 +194,24 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
...
         '^is_executable$' => null,
-        '^is_file$' => null,
+        '^is_file$' => '\Magento\Framework\Filesystem\Driver\File->isFile()',
```
Comment (ihor-sviziev) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]:
```suggestion
        '^is_file$' => '\Magento\Framework\Filesystem\Driver\File->isFile',
```

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 10]
```diff
@@ -194,24 +194,24 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
...
         '^is_link$' => null,
-        '^is_readable$' => null,
-        '^is_writable$' => null,
-        '^is_writeable$' => null,
+        '^is_readable$' => '\Magento\Framework\Filesystem\Driver\File->isReadable()',
```
Comment (ihor-sviziev) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]:
```suggestion
        '^is_readable$' => '\Magento\Framework\Filesystem\Driver\File->isReadable',
```

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 11]
```diff
@@ -194,24 +194,24 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
...
+        '^is_writable$' => 'Magento\Framework\Filesystem\Io\File->isWriteable()',
```
Comment (ihor-sviziev) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]:
```suggestion
        '^is_writable$' => 'Magento\Framework\Filesystem\Io\File->isWriteable',
```

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 12]
```diff
@@ -194,24 +194,24 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
...
+        '^is_writeable$' => 'Magento\Framework\Filesystem\Io\File->isWriteable()',
```
Comment (ihor-sviziev) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]:
```suggestion
        '^is_writeable$' => 'Magento\Framework\Filesystem\Io\File->isWriteable',
```

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 13]
```diff
@@ -194,24 +194,24 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
...
         '^is_uploaded_file$' => null,
         '^glob$' => null,
         '^ssh2_.*$' => null,
-        '^delete$' => null,
+        '^delete$' => '\Magento\Framework\Filesystem\Driver\File->deleteFile()',
```
Comment (ihor-sviziev) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]:
```suggestion
        '^delete$' => '\Magento\Framework\Filesystem\Driver\File->deleteFile',
```

**Comment by diazwatson** [comment]
@ihor-sviziev all requested changes are now done.
I have also added some functions from `\Magento\Framework\Filesystem\Driver\File`

**Comment by lenaorobei** [comment]
@buskamuza could you please have a look at these recommendations? It looks like we need to suggest something different from `Magento\Framework\Filesystem\Io`  because it goes against this proposal https://github.com/magento/architecture/issues/219.

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 14]
```diff
@@ -79,7 +79,7 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
         '^link$' => null,
         '^mail$' => null,
         '^mb_send_mail$' => null,
-        '^mkdir$' => null,
+        '^mkdir$' => 'Magento\Framework\Filesystem\Io\File->mkdir',
```
Comment (buskamuza) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]: `\Magento\Framework\Filesystem\DriverInterface::createDirectory()`

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 15]
```diff
@@ -39,11 +39,11 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
         '^call_user_func_array$' => null,
         '^chdir$' => null,
         '^chgrp$' => null,
-        '^chmod$' => null,
+        '^chmod$' => 'Magento\Framework\Filesystem\Io\File->chmod',
```
Comment (buskamuza) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]: `\Magento\Framework\Filesystem\DriverInterface::changePermissions() or \Magento\Framework\Filesystem\DriverInterface::changePermissionsRecursively()`

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 16]
```diff
@@ -106,14 +106,14 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
...
         '^readgzfile$' => null,
-        '^readline$' => null,
+        '^readline$' => '\Magento\Framework\Filesystem\Driver\File->fileReadLine',
         '^readlink$' => null,
         '^register_shutdown_function$' => null,
         '^register_tick_function$' => null,
-        '^rename$' => null,
-        '^rmdir$' => null,
+        '^rename$' => '\Magento\Framework\Filesystem\Driver\File->raname',
+        '^rmdir$' => 'Magento\Framework\Filesystem\Io\File->rmdir',
```
Comment (buskamuza) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]: `\Magento\Framework\Filesystem\DriverInterface::deleteDirectory()`

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 17]
```diff
@@ -194,24 +194,24 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
...
         '^is_executable$' => null,
-        '^is_file$' => null,
+        '^is_file$' => '\Magento\Framework\Filesystem\Driver\File->isFile',
         '^is_link$' => null,
-        '^is_readable$' => null,
-        '^is_writable$' => null,
-        '^is_writeable$' => null,
+        '^is_readable$' => '\Magento\Framework\Filesystem\Driver\File->isReadable',
+        '^is_writable$' => 'Magento\Framework\Filesystem\Io\File->isWriteable',
+        '^is_writeable$' => 'Magento\Framework\Filesystem\Io\File->isWriteable',
```
Comment (buskamuza) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]: `\Magento\Framework\Filesystem\DriverInterface::isWritable()`

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 18]
```diff
@@ -53,17 +53,17 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
         '^dcngettext$' => null,
         '^dgettext$' => null,
         '^dio_.*$' => null,
-        '^dirname$' => null,
+        '^dirname$' => 'Magento\Framework\Filesystem\Io\File->dirname',
```
Comment (buskamuza) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]: `\Magento\Framework\Filesystem\DriverInterface::getParentDirectory()`

**Comment by buskamuza** [comment]
Thanks, @lenaorobei . The proposal is not ready yet, but at least we should avoid recommending `Io` library.

**Comment by diazwatson** [comment]
Thanks @buskamuza @lenaorobei  I'll update this PR today

**Comment by diazwatson** [comment]
@buskamuza requested changes are done, please have a look when you get a chance

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 19]
```diff
@@ -39,11 +39,11 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
         '^call_user_func_array$' => null,
         '^chdir$' => null,
         '^chgrp$' => null,
-        '^chmod$' => null,
+        '^chmod$' => '\Magento\Framework\Filesystem\DriverInterface::changePermissions() or \Magento\Framework\Filesystem\DriverInterface::changePermissionsRecursively()',
```
Comment (lenaorobei) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]: Need to fix `Line exceeds maximum limit of 120 characters; contains 171 characters` here.

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 20]
```diff
@@ -39,11 +39,12 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
...
         '^chdir$' => null,
         '^chgrp$' => null,
-        '^chmod$' => null,
+        '^chmod$' => '\Magento\Framework\Filesystem\DriverInterface::changePermissions() 
+        or \Magento\Framework\Filesystem\DriverInterface::changePermissionsRecursively()',
         '^chown$' => null,
         '^chroot$' => null,
         '^com_load_typelib$' => null,
-        '^copy$' => null,
+        '^copy$' => '\Magento\Framework\Filesystem\Driver\File::copy()',
```
Comment (lenaorobei) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]:
```suggestion
        '^copy$' => 'Magento\Framework\Filesystem\DriverInterface::copy()',
```

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 21]
```diff
@@ -53,17 +54,17 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
         '^dcngettext$' => null,
         '^dgettext$' => null,
         '^dio_.*$' => null,
-        '^dirname$' => null,
+        '^dirname$' => '\Magento\Framework\Filesystem\DriverInterface::getParentDirectory()',
```
Comment (lenaorobei) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]:
```suggestion
        '^dirname$' => 'Magento\Framework\Filesystem\DriverInterface::getParentDirectory()',
```

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 22]
```diff
@@ -53,17 +54,17 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
...
         '^dngettext$' => null,
         '^domxml_.*$' => null,
         '^fbsql_.*$' => null,
         '^fdf_add_doc_javascript$' => null,
         '^fdf_open$' => null,
-        '^fopen$' => null,
-        '^fclose$' => null,
+        '^fopen$' => '\Magento\Framework\Filesystem\Driver\File::fileOpen()',
```
Comment (lenaorobei) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]:
```suggestion
        '^fopen$' => 'Magento\Framework\Filesystem\DriverInterface::fileOpen()',
```

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 23]
```diff
@@ -53,17 +54,17 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
...
+        '^fclose$' => '\Magento\Framework\Filesystem\Driver\File::fileClose()',
```
Comment (lenaorobei) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]:
```suggestion
        '^fclose$' => 'Magento\Framework\Filesystem\DriverInterface::fileClose()',
```

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 24]
```diff
@@ -53,17 +54,17 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
...
         '^fsockopen$' => null,
         '^ftp_.*$' => null,
-        '^fwrite$' => null,
+        '^fwrite$' => '\Magento\Framework\Filesystem\Driver\File::fileWrite()',
```
Comment (lenaorobei) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]:
```suggestion
        '^fwrite$' => '\Magento\Framework\Filesystem\Driver\File::fileWrite()',
```

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 25]
```diff
@@ -96,7 +97,7 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
         '^parse_str$' => null,
         '^parse_url$' => null,
         '^parsekit_compile_string$' => null,
-        '^pathinfo$' => null,
+        '^pathinfo$' => 'Magento\Framework\Filesystem\Io\File::getPathInfo->()',
```
Comment (lenaorobei) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]:
```suggestion
        '^pathinfo$' => null,
```

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 26]
```diff
@@ -79,7 +80,7 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
         '^link$' => null,
         '^mail$' => null,
         '^mb_send_mail$' => null,
-        '^mkdir$' => null,
+        '^mkdir$' => '\Magento\Framework\Filesystem\DriverInterface::createDirectory()',
```
Comment (lenaorobei) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]:
```suggestion
        '^mkdir$' => 'Magento\Framework\Filesystem\DriverInterface::createDirectory()',
```

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 27]
```diff
@@ -106,14 +107,14 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
         '^print_r$' => null,
         '^printf$' => null,
         '^putenv$' => null,
-        '^readfile$' => null,
+        '^readfile$' => '\Magento\Framework\Filesystem\Driver\File::fileRead()',
```
Comment (lenaorobei) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]:
```suggestion
        '^readfile$' => 'Magento\Framework\Filesystem\DriverInterface::fileRead()',
```

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 28]
```diff
@@ -106,14 +107,14 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
...
         '^readgzfile$' => null,
-        '^readline$' => null,
+        '^readline$' => '\Magento\Framework\Filesystem\Driver\File::fileReadLine()',
```
Comment (lenaorobei) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]:
```suggestion
        '^readline$' => 'Magento\Framework\Filesystem\DriverInterface::fileReadLine()',
```

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 29]
```diff
@@ -126,9 +127,9 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
         '^socket_.*$' => null,
         '^stream_.*$' => null,
         '^sybase_.*$' => null,
-        '^symlink$' => null,
+        '^symlink$' => '\Magento\Framework\Filesystem\Driver\File::symlink()',
```
Comment (lenaorobei) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]:
```suggestion
        '^symlink$' => 'Magento\Framework\Filesystem\DriverInterface::symlink()',
```

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 30]
```diff
@@ -106,14 +107,14 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
...
+        '^readfile$' => '\Magento\Framework\Filesystem\Driver\File::fileRead()',
         '^readgzfile$' => null,
-        '^readline$' => null,
+        '^readline$' => '\Magento\Framework\Filesystem\Driver\File::fileReadLine()',
         '^readlink$' => null,
         '^register_shutdown_function$' => null,
         '^register_tick_function$' => null,
-        '^rename$' => null,
-        '^rmdir$' => null,
+        '^rename$' => '\Magento\Framework\Filesystem\Driver\File::raname()',
```
Comment (lenaorobei) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]:
```suggestion
        '^rename$' => 'Magento\Framework\Filesystem\DriverInterface::raname()',
```

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 31]
```diff
@@ -106,14 +107,14 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
...
+        '^rmdir$' => '\Magento\Framework\Filesystem\DriverInterface::deleteDirectory()',
```
Comment (lenaorobei) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]:
```suggestion
        '^rmdir$' => 'Magento\Framework\Filesystem\DriverInterface::deleteDirectory()',
```

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 32]
```diff
@@ -126,9 +127,9 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
         '^socket_.*$' => null,
         '^stream_.*$' => null,
         '^sybase_.*$' => null,
-        '^symlink$' => null,
+        '^symlink$' => '\Magento\Framework\Filesystem\Driver\File::symlink()',
         '^syslog$' => null,
-        '^touch$' => null,
+        '^touch$' => '\Magento\Framework\Filesystem\Driver\File::touch()',
```
Comment (lenaorobei) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]:
```suggestion
        '^touch$' => 'Magento\Framework\Filesystem\DriverInterface::touch()',
```

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 33]
```diff
@@ -149,7 +150,7 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
         '^fdf_.*$' => null,
         '^fget.*$' => null,
         '^fread$' => null,
-        '^fflush$' => null,
+        '^fflush$' => '\Magento\Framework\Filesystem\Driver\File::fileFlush()',
```
Comment (lenaorobei) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]:
```suggestion
        '^fflush$' => 'Magento\Framework\Filesystem\DriverInterface::fileFlush()',
```

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 34]
```diff
@@ -194,24 +195,24 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
         '^gettype$' => null,
         '^var_dump$' => null,
         '^tempnam$' => null,
-        '^realpath$' => null,
+        '^realpath$' => '\Magento\Framework\Filesystem\Driver\File::getRealPath()',
```
Comment (lenaorobei) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]:
```suggestion
        '^realpath$' => 'Magento\Framework\Filesystem\DriverInterface::getRealPath()',
```

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 35]
```diff
@@ -194,24 +195,24 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
...
         '^linkinfo$' => null,
-        '^lstat$' => null,
+        '^lstat$' => '\Magento\Framework\Filesystem\Driver\File::stat()',
```
Comment (lenaorobei) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]:
```suggestion
        '^lstat$' => 'Magento\Framework\Filesystem\DriverInterface::stat()',
```

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 36]
```diff
@@ -194,24 +195,24 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
...
         '^stat$' => null,
         '^lchgrp$' => null,
         '^lchown$' => null,
         '^show_source$' => null,
-        '^is_dir$' => null,
+        '^is_dir$' => '\Magento\Framework\Filesystem\Driver\File::isDirectory()',
```
Comment (lenaorobei) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]:
```suggestion
        '^is_dir$' => 'Magento\Framework\Filesystem\DriverInterface::isDirectory()',
```

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 37]
```diff
@@ -194,24 +195,24 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
...
         '^is_executable$' => null,
-        '^is_file$' => null,
+        '^is_file$' => '\Magento\Framework\Filesystem\Driver\File::isFile()',
```
Comment (lenaorobei) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]:
```suggestion
        '^is_file$' => 'Magento\Framework\Filesystem\DriverInterface::isFile()',
```

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 38]
```diff
@@ -194,24 +195,24 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
...
         '^is_link$' => null,
-        '^is_readable$' => null,
-        '^is_writable$' => null,
-        '^is_writeable$' => null,
+        '^is_readable$' => '\Magento\Framework\Filesystem\Driver\File::isReadable()',
```
Comment (lenaorobei) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]:
```suggestion
        '^is_readable$' => 'Magento\Framework\Filesystem\DriverInterface::isReadable()',
```

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 39]
```diff
@@ -194,24 +195,24 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
...
+        '^is_writable$' => '\Magento\Framework\Filesystem\DriverInterface::isWritable()',
```
Comment (lenaorobei) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]:
```suggestion
        '^is_writable$' => 'Magento\Framework\Filesystem\DriverInterface::isWritable()',
```

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 40]
```diff
@@ -194,24 +195,24 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
...
+        '^is_writeable$' => '\Magento\Framework\Filesystem\DriverInterface::isWritable()',
```
Comment (lenaorobei) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]:
```suggestion
        '^is_writeable$' => 'Magento\Framework\Filesystem\DriverInterface::isWritable()',
```

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 41]
```diff
@@ -194,24 +195,24 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
...
         '^is_uploaded_file$' => null,
-        '^glob$' => null,
+        '^glob$' => '\Magento\Framework\Filesystem\Glob::glob()',
```
Comment (lenaorobei) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]:
```suggestion
        '^glob$' => 'Magento\Framework\Filesystem\Glob::glob()',
```

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 42]
```diff
@@ -194,24 +195,24 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
...
         '^ssh2_.*$' => null,
-        '^delete$' => null,
+        '^delete$' => '\Magento\Framework\Filesystem\Driver\File::deleteFile()',
```
Comment (lenaorobei) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]:
```suggestion
        '^delete$' => 'Magento\Framework\Filesystem\DriverInterface::deleteFile()',
```

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 43]
```diff
@@ -39,11 +39,12 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
         '^call_user_func_array$' => null,
         '^chdir$' => null,
         '^chgrp$' => null,
-        '^chmod$' => null,
+        '^chmod$' => '\Magento\Framework\Filesystem\DriverInterface::changePermissions()
```
Comment (lenaorobei) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]:
```suggestion
        '^chmod$' => 'Magento\Framework\Filesystem\DriverInterface::changePermissions() 
```

File: Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php [hunk 44]
```diff
@@ -39,11 +39,12 @@ class DiscouragedFunctionSniff extends ForbiddenFunctionsSniff
         '^call_user_func_array$' => null,
         '^chdir$' => null,
         '^chgrp$' => null,
-        '^chmod$' => null,
+        '^chmod$' => '\Magento\Framework\Filesystem\DriverInterface::changePermissions() 
+        or \Magento\Framework\Filesystem\DriverInterface::changePermissionsRecursively()',
```
Comment (lenaorobei) [inline Magento2/Sniffs/Functions/DiscouragedFunctionSniff.php]:
```suggestion
        or Magento\Framework\Filesystem\DriverInterface::changePermissionsRecursively()',
```

**Pull request merged**